### DISCUSSION TITLE
Nightly releases

### DISCUSSION THREAD

**Discussion by Joozty** [explosion/spaCy#7192] [New Features & Project Ideas]
I was wondering. What do you think about adding support for nightly releases? I can see you already use a package [spacy-nightly](https://pypi.org/project/spacy-nightly/) but it seems it's not updated regularly.  I can imagine pre-release spacy every night (or based on a different heuristic). 

It would help to use new features/fixes straight ahead. Let's imagine you fixed some bug that was stopping me in my work and I would like to use your fix.  Now I have to option: wait for a release or build from the [source](https://spacy.io/usage#source) which is not really convenient (at least for me 😄) Also I can see one big advantage. You can get instant feedback from the users that the reported bug is really fixed. (hopefully, it's converted by tests but still)

What do you think guys?

**Comment by ines** [comment]
Hi! I've actually been thinking about this before and how it'd be cool to automatically publish a release from the latest state on `master` once a day or so. It would even be super useful for us internally as well 😅 We even have our own build system in place that takes care of building the sdist and wheel and making it available for release: https://github.com/explosion/wheelwright/

We'll need to think about this some more in detail, but the process could look like this:

1. Automatically increment the version in `about.py` (e.g. `.dev0` &rarr; `.dev1` and so on).
2. Start a `wheelwright` build from the latest commit on the `master` branch.
3. If the build succeeds, auto-release that to `spacy-nightly`. (We don't have the auto-release implemented as an option for `wheelwright` on Azure Pipelines yet, but it's something we've wanted to have for a while and it should be easy to add). If the build fails, alert us on Slack.

We could probably do something like this with GitHub actions? 🤔
